tests/functional/dependencies/test_local_dependency.py

```diff
@@ -208,9 +207,8 @@ def models(self):
 
     def test_missing_dependency(self, project):
         # dbt should raise a runtime exception
-        res = run_dbt(["compile"], expect_pass=False)
-        assert len(res) == 1
-        assert res[0].status == RunStatus.Error
```

Comment:
reverts the changes made here: https://github.com/dbt-labs/dbt-core/pull/8066/files#diff-4f5f78d9b3b6f6eda81916de152b5c17b9d1ef14e40b7a2031c4d10f601d6474L210-L211

On closer inspection, I'm not sure why this test was modified, it is a change in behaviour for non fail-fast functionality.